cookbook/teaser-provider.rst

```diff
@@ -0,0 +1,120 @@
+Provider for Teaser-Selection
+=============================
+
+TeaserProvider are used to load data for Teaser-Selection. It returns an array
+of Teaser instances by an array of ids. This array can be configured by a
+content-type in the Sulu-Admin.
+
+The Teaser-Object consists of following properties:
+
+* id
+* type (e.g. content or article)
+* locale
+* title
+* description
+* moreText (link-text)
+* mediaId
+* url
+
+Example
+-------
+
+This example assumes that the entities will be select by a datagrid.
+For this we already build a abstract implementation which can be
+configured wil the `TeaserConfiguration`.
+
+.. code-block:: php
+
+    namespace AppBundle\Teaser;
+
+    use Sulu\Bundle\ContentBundle\Teaser\Configuration\TeaserConfiguration;
+    use Sulu\Bundle\ContentBundle\Teaser\Provider\TeaserProviderInterface;
+    use Sulu\Bundle\ContentBundle\Teaser\Teaser;
+
+    class TeaserProvider implements TeaserProviderInterface
+    {
+        public function getConfiguration()
+        {
+            return new TeaserConfiguration(
+                'app.teaser',
+                'teaser-selection/list@sulucontent',
+                [
+                    'url' => '...', // your api url
+                    'resultKey' => '...', // your api result-key
+                    'searchFields' => ['title'],
+                    'matchings' => [
+                        [
+                            'content' => 'public.title',
+                            'name' => 'title',
+                        ],
+                    ],
+                ]
+            );
+        }
+
+        public function find(array $ids, $locale)
+        {
+            if (0 === count($ids)) {
+                return [];
+            }
+
+            $items = ...; // load items by id
+            foreach ($items as $item) {
+                $result[] = new Teaser(...); // create teaser foreach item
+            }
+
+            return $result;
+        }
+    }
+
+Now you can create a service for this class and add the tag
+`<tag name="sulu.teaser.provider" alias="<your teaser-type>"/>`.
```

Comment:
Maybe we should another notation than `<your teaser-type>`, because the brackets already used in this context.